### PR TITLE
boards/ikea-tradfri: add board documentation.

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description
This adds board documentation for the IKEA TRÅDFRI module.

### Testing procedure
Not applicable.

### Issues/PRs references
None.

### CODE REVIEW

File: boards/ikea-tradfri/doc.txt [hunk 1]
```diff
@@ -2,4 +2,169 @@
...
+
+## Hardware
+
+### MCU
+| MCU             | EFR32MG1P132F256GM32                                                                         |
+|-----------------|----------------------------------------------------------------------------------------------|
+| Family          | ARM Cortex-M4F                                                                               |
+| Vendor          | Silicon Labs                                                                                 |
+| Vendor Family   | EFR32 Mighty Gecko 1P                                                                        |
+| RAM             | 31.0KB                                                                                       |
```
Comment (aabadie) [inline boards/ikea-tradfri/doc.txt]: The doc says 32KB. Is this a typo ?

```suggestion
| RAM             | 31.0KB                                                                                       |
```

File: boards/ikea-tradfri/doc.txt [hunk 2]
```diff
@@ -2,4 +2,169 @@
...
+| Flash           | 256.0KB                                                                                      |
+| EEPROM          | no                                                                                           |
+| Frequency       | up to 38.4 MHz                                                                               |
+| FPU             | yes                                                                                          |
+| MPU             | yes                                                                                          |
+| DMA             | 12 channels                                                                                  |
+| Timers          | 2x 16-bit + 1x 16-bit (low power)                                                            |
+| ADCs            | 12-bit ADC                                                                                   |
+| UARTs           | 3x UART, 2x USART, 1x LEUART                                                                 |
+| SPIs            | 2x USART                                                                                     |
```
Comment (aabadie) [inline boards/ikea-tradfri/doc.txt]:
```suggestion
| SPIs            | 2x                                                                                           |
```

File: boards/ikea-tradfri/doc.txt [hunk 3]
```diff
@@ -2,4 +2,169 @@
...
+| RAM             | 31.0KB                                                                                       |
+| Flash           | 256.0KB                                                                                      |
+| EEPROM          | no                                                                                           |
+| Frequency       | up to 38.4 MHz                                                                               |
+| FPU             | yes                                                                                          |
+| MPU             | yes                                                                                          |
+| DMA             | 12 channels                                                                                  |
+| Timers          | 2x 16-bit + 1x 16-bit (low power)                                                            |
+| ADCs            | 12-bit ADC                                                                                   |
+| UARTs           | 3x UART, 2x USART, 1x LEUART                                                                 |
```
Comment (aabadie) [inline boards/ikea-tradfri/doc.txt]:
```suggestion
| UARTs           | 3x UART: 2x USART, 1x LEUART                                                                 |
```

File: boards/ikea-tradfri/doc.txt [hunk 4]
```diff
@@ -2,4 +2,169 @@
...
+| SPIs            | 2x USART                                                                                     |
+| I2Cs            | 1x                                                                                           |
+| Vcc             | 1.85V - 3.8V                                                                                 |
+| Datasheet       | [Datasheet](https://www.silabs.com/Support%20Documents/TechnicalDocs/EFR32MG1-DataSheet.pdf) |
+| Manual          | [Manual](http://www.silabs.com/Support%20Documents/TechnicalDocs/EFR32-ReferenceManual.pdf)  |
```
Comment (aabadie) [inline boards/ikea-tradfri/doc.txt]: This url is broken

File: boards/ikea-tradfri/doc.txt (hunk 1 above)
Comment (basilfx) [reply boards/ikea-tradfri/doc.txt]: Technically, it's correct. But when you use the radio library (RAIL), it uses 1KiB of RAM.

This is the same as with the SLTB001a.

File: boards/ikea-tradfri/doc.txt (hunk 3 above)
Comment (basilfx) [reply boards/ikea-tradfri/doc.txt]: Fixed. I accidentally copied this information wrong. It only has 2x USART.

File: boards/ikea-tradfri/doc.txt (hunk 4 above)
Comment (basilfx) [reply boards/ikea-tradfri/doc.txt]: Fixed URLs.

File: boards/ikea-tradfri/doc.txt (hunk 1 above)
Comment (basilfx) [reply boards/ikea-tradfri/doc.txt]: Changed this into 32.0KB (1.0KB reserved). That's what the linker script uses.

**Comment by basilfx** [comment]
@aabadie Updated this PR with some more fixes. I will use this format for the other boards as well.

**Pull request merged**